src/commands.rs

```diff
@@ -48,6 +50,11 @@ pub struct CliOpts {
         default_value = "testnet"
     )]
     pub network: Network,
+    /// Sets the wallet data directory.
```

Comment:
Good question.. While I have never thought about this before, but I think running bdk-cli with memory db doesn't make much sense. 

Each bdk-cli call starts a wallet, does operations, then drops the wallet. This approach only works when the wallet state is persisted in a db and the new wallet is loaded from the db, in subsequent calls, identified by the descriptor checksum value (unless use gives a name, then they need call bdk-cli methods with the same wallet name).

In case of memory db every call will be a fresh new wallet.. So you can't do much with it.

Reply:
OHHHHH you're right. Which means that #114 might not make sense.

But... I'm not sure if I can use the other dbs on WASM. I'll think more about this, if this PR is already merged when I find a solution I'll just update the code myself :)

Reply:
Oh drat! I didn't see this before merging #114. 😬 It sounds like we may still need memory db support for wasm, or could also be used with the cli in repl mode.  But if you want me to revert #114 let me know.

Reply:
I verified using the memory db with `repl` mode:
```shell
cargo run --no-default-features --features esplora-ureq,repl -- repl --descriptor "wpkh(tpubEBr4i6yk5nf5DAaJpsi9N2pPYBeJ7fZ5Z9rmN4977iYLCGco1VyjB9tvvuvYtfZzjD5A8igzgw3HeWeeKFmanHYqksqZXYXGsw5zjnj7KM9/*)"
```
It works fine, but with a couple build warnings:
```shell
warning: unused variable: `database_path`
   --> src/utils.rs:169:9
    |
169 |     let database_path = prepare_wallet_db_dir(wallet_name)?;
    |         ^^^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_database_path`
    |
    = note: `#[warn(unused_variables)]` on by default

warning: variable does not need to be mutable
   --> src/utils.rs:131:9
    |
131 |     let mut db_dir = prepare_home_wallet_dir(wallet_name)?;
    |         ----^^^^^^
    |         |
    |         help: remove this `mut`
    |
    = note: `#[warn(unused_mut)]` on by default
```

Reply:
I am very confused.. Did the wallet work worked in repl mode?? Like, could you send sats to it, sync and get the balance?? 

I think there might be few ways to make it only work in `repl` mode. But as the current code stands, it should not work.. Or I am missing something important..

Reply:
The warnings happen because the code is written with a "always db" as assumption.. If we need memory db for wasm we might need to modify the code a bit, along with the build script.. Getting bdk-cli into wasm should be an important enough task to do whatever is needed..

Reply:
> I am very confused.. Did the wallet work worked in repl mode with memory db?? Like, could you send sats to it, sync and get the balance?? It might not throw any error, but it won't be functional wallet if wallet data gets wiped out after every call..

Yes the cli wallet works with a memory db in repl mode using the above feature settings. I verified syncing, receiving sats from a testnet faucet, list_unspent, get_balance. The reason it works is because in repl mode the wallet (and memory db) is created before you enter the repl loop.

Reply:
Ahh right.. Its not the command where the iteration happens, its the loop.. Thanks.. Then it works.. #114 seems good to me.. So we just need to clear up few things to make room for memory db in the code.. Also should we only allow `repl` mode while using memory db?

Reply:
I don't think we need to limit using the mem db to only repl mode. I can imagine a wasm web wallet that isn't using the repl feature, but does work like repl with a single Wallet instance that it sync's on startup and keeps live until page is closed. We already make sqlite-db a default feature so most casual users shouldn't accidentally enabled the memory db.

Reply:
Okay.. Ya if that works then it would be sweet.. I will rebase this after #115 . I think that one is clean to go too.. And update to get memory db without the warnings..